libs/template/src/test-setup.ts

```diff
@@ -2,3 +2,16 @@ import 'jest-preset-angular/setup-jest';
 
 /* @Notice: schematics have long-running tests that timeout if no cache hit. */
 jest.setTimeout(100_000);
+
+import { getTestBed } from '@angular/core/testing';
+import {
+  BrowserDynamicTestingModule,
+  platformBrowserDynamicTesting,
+} from '@angular/platform-browser-dynamic/testing';
+
+getTestBed().resetTestEnvironment();
+getTestBed().initTestEnvironment(
+  BrowserDynamicTestingModule,
+  platformBrowserDynamicTesting(),
+  { teardown: { destroyAfterEach: false } }
+);
```

Comment:
suggestion: I know this is the default Nx migration but it would be better if our unit tests could run with the default TestBed setup where `destroyAfterEach` is set to `true` by removing all the added lines
```suggestion
```